core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -405,4 +405,103 @@ default RelDataType deriveDecimalModType(RelDataTypeFactory typeFactory,
     return null;
   }
 
+  /**
+   * Infers the return type of a decimal truncate operation. Decimal truncate
+   * involves at least one decimal operand.
+   *
+   * <p>The default implementation is SQL:2003 compliant: the declared type of
+   * the result is the declared type of the first operand (decimal to be truncated).
+   *
+   * @see Glossary#SQL2003 SQL:2003 Part 2 Section 6.27
+   *
+   * <p>Rules:
+   *
+   * <ul>
+   * <li>Let p1, s1 be the precision and scale of the first operand</li>
+   * <li>Let s2 be the scale value to truncate</li>
+   * <li>Let p, s be the precision and scale of the result</li>
+   * <li>Let d be the number of whole digits in the result</li>
+   * <li>Then the result type is a decimal with:
```

Comment:
Only `p` `s` `p1` `s2`  are used, remove the useless comments.

Reply:
removed done.